### PR TITLE
Corrections to infomax.py

### PR DESCRIPTION
**Pull request by amanyazevedoamin**
- Scale pixel values for input images to between 0 and 1 (i.e order of magnitude of the weights). This prevents system explosion to infinity and allows a default learning rate of 0.01
- Scaling factor to prevent saturation of the tanh function. Note that this is essentially a hyper parameter, may require optimising.
- Replacement of element wise multiplications with matrix multiplications

### CODE REVIEW

File: navbench/infomax.py [hunk 1]
```diff
@@ -30,17 +30,34 @@ def __init__(self, num_inputs, num_hidden=None, learning_rate=DEFAULT_LEARNING_R
...
+        self.imageScaling=self.tanhScalingFactor*self.greyscaleScalingFactor
+        
 
     def train(self, image):
-        u = self.weights * image.ravel()
+        
+        
+        u=np.matmul(self.weights,image.ravel()/self.imageScaling)
+
         y = np.tanh(u)
```
Comment (alexdewar) [inline navbench/infomax.py:49]: I'd just use ``tanhScalingFactor`` directly here for clarity:
```py
y = np.tanh(self.tanhScalingFactor * u)
```

(I think this is equivalent to what you've done, unless I've missed something....)

File: navbench/infomax.py [hunk 2]
```diff
@@ -30,17 +30,34 @@ def __init__(self, num_inputs, num_hidden=None, learning_rate=DEFAULT_LEARNING_R
...
+        self.greyscaleScalingFactor=255
+        
+        self.imageScaling=self.tanhScalingFactor*self.greyscaleScalingFactor
+        
 
     def train(self, image):
-        u = self.weights * image.ravel()
+        
+        
+        u=np.matmul(self.weights,image.ravel()/self.imageScaling)
```
[the remark this answers is not in the document]
Comment (alexdewar) [reply navbench/infomax.py:47]: (This would also mean you could drop the ``greyscaleScalingFactor`` and ``imageScaling`` member variables.)

[the remark this answers is not in the document]
Comment (alexdewar) [reply navbench/infomax.py:47]: Sorry, what I actually meant was convert to float and normalise to be between 0 and 1. Generally images made of floats/doubles *always* range between 0 and 1 (e.g. OpenCV will normalise this way by default if you convert from a ``uchar`` matrix to a ``float`` one), so any inputs should be like that unless I've screwed up somewhere.

If we wanted to be extra careful/pedantic, we could also stick a check in to verify that ``image`` is between 0 and 1.

File: navbench/infomax.py (hunk 1 above)
Comment (alexdewar) [reply navbench/infomax.py:49]: Ah yes, oops! But I agree that a multiplication probably is more intuitive

File: navbench/infomax.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (amanyazevedoamin) [reply navbench/infomax.py:47]: Makes sense - pytorch also does this when you convert the image to tensor. Shall we stick a check in? Or leave it so that there's a bit more flexibility? Happy to go with what you think, as you have a better idea of where and how this code is being used.
